src/persist-client/src/internal/encoding.rs

```diff
@@ -1289,10 +1289,29 @@ impl LazyPartStats {
     /// This does not cache the returned value, it decodes each time it's
     /// called.
     pub fn decode(&self) -> PartStats {
-        let key = self.key.decode().expect("valid proto");
-        PartStats {
-            key: key.into_rust().expect("valid stats"),
-        }
+        // These clones are not ideal, but we're only doing it while debugging a
+        // pesky panic.
+        let key_cloned = self.key.clone();
```

Comment:
Do we need this clone? This method is `&self` so I don't think anything can take ownership of `self.key`